x-pack/libbeat/common/cloudfoundry/config.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package cloudfoundry
+
+import (
+	"crypto/tls"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common/transport/tlscommon"
+
+	"github.com/gofrs/uuid"
+)
+
+type Config struct {
+	// CloudFoundry credentials for retrieving OAuth tokens
+	ClientID     string `config:"client_id" validate:"required"`
+	ClientSecret string `config:"client_secret" validate:"required"`
+
+	// TLS configuration for the client
+	TLS *tlscommon.Config `config:"ssl"`
+
+	// Override URLs returned from the CF client
+	APIAddress     string `config:"api_address"`
+	DopplerAddress string `config:"doppler_address"`
+	UaaAddress     string `config:"uaa_address"`
+	RlpAddress     string `config:"rlp_address"`
+
+	// ShardID when retrieving events from loggregator, sharing this ID across
+	// multiple filebeats will shard the load of receiving and sending events.
+	ShardID string `config:"shard_id"`
+
+	// Maximum amount of time to cache application objects from CF client
+	CacheDuration time.Duration `config:"cache_duration"`
+}
+
+// InitDefaults initialize the defaults for the configuration.
+func (c *Config) InitDefaults() {
```

Comment:
Not a strong opinion, but we normally use this pattern: https://github.com/elastic/beats/blob/837279a4b5c2b5e560bfc8fc3e6dd7d4c1cebd12/metricbeat/helper/config.go#L35
so you can get defaults on a single call, this way you don't risk forgetting to call InitDefaults

Reply:
I think this is a new way of initializing defaults :slightly_smiling_face: https://github.com/elastic/go-ucfg/pull/141

Reply:
Awesome, I missed this change :+1: